demo/freedv_700d_rx.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+'''
+  Demo receive program for FreeDV API 700D mode.
+
+  cd ~/codec2/build_linux
+  cat ../raw/ve9qrp_10s.raw | ./demo/freedv_700d_tx | ../demo/freedv_700d_rx.py | aplay -f S16_LE
+
+  Credits: Thanks DJ2LS, xssfox, VK5QI
+'''
+
+import ctypes
+from ctypes import *
+import sys
+import pathlib
+
+libname = pathlib.Path().absolute() / "src/libcodec2.so"
+c_lib = ctypes.CDLL(libname)
+c_lib.freedv_open.restype = ctypes.POINTER(ctypes.c_ubyte)
+c_lib.freedv_get_n_max_speech_samples.restype = c_int
+c_lib.freedv_nin.restype = c_int
+
+FREEDV_MODE_700D = 7 # from freedv_api.h             
+freedv = c_lib.freedv_open(FREEDV_MODE_700D)
+n_max_speech_samples = c_lib.freedv_get_n_max_speech_samples(freedv)
+speech_out = (ctypes.c_short * n_max_speech_samples)()
+speech_out = bytes(speech_out)
+
+while True:
+    nin = c_lib.freedv_nin(freedv)
+    demod_in = sys.stdin.buffer.read(nin*2)
+    if len(demod_in) == 0: quit()
+    nout = c_lib.freedv_rx(freedv, speech_out, demod_in)
+    sys.stdout.buffer.write(speech_out[:nout*2])
```

Comment:
@DJ2LS @xssfox @darksidelemm :point_up: 

Any suggestions on how to improve this or make it simpler?

Reply:
I would suggest adding a
"import time" at the top section 
and then 
"time.sleep(0.01)" inside while True: to reduce  CPU load a little bit. while True consumes all it gets from the CPU.

I'm not at home right now - sorry for code inside the comment area

Reply:
I tried running it with no I/O and it doesn't seem to use any CPU.  I think it's blocking on the stdin/stdout I/O so just sits there not using CPU :slightly_smiling_face: 

You're other suggestions (via email) worked well.  Program is even simpler now.  I'm not sure how Python works out the types, but it runs OK